File: java/src/microbiosima/Individual.java
/*******************************************************************************
 *
 * Copyright (C) 2014, 2015 Qinglong Zeng, Jeet Sukumaran, Steven Wu and Allen Rodrigo
 *
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *******************************************************************************/
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package microbiosima;

/**
 *
 * @author John
 */
public class Individual {

	private double[] microbiome;//NOTE: is that possible to use int[]
	
	private int numberEnvironmentalSpecies;
	private int numberMicrobePerHost;
	
	public Individual(String[] initial_microbiome, int nomph, int noes) {
		numberEnvironmentalSpecies = noes;
		microbiome = new double[numberEnvironmentalSpecies];
		for (int i = 0; i < numberEnvironmentalSpecies; i++) {
			microbiome[i] = Double.parseDouble(initial_microbiome[i]);
		}
		numberMicrobePerHost = nomph;
	}

	public String microbial_sequences() {
		char[] microbiome_sequence = new char[numberEnvironmentalSpecies];
		for (int i = 0; i < numberEnvironmentalSpecies; i++) {
			if (microbiome[i] > 0) {
				microbiome_sequence[i] = '1';
			} else {
				microbiome_sequence[i] = '0';
			}
		}
		return new String(microbiome_sequence);
	}

	/**
	 * @return the microbiome
	 */
	public double[] getMicrobiome() {
		return microbiome;
	}
//	public double getMicrobiome(int i) {
//		return microbiome[i];
//	}
}

File: python/microbiosima/individual.py
################################################################################
#
# Copyright (C) 2014, 2015 Qinglong Zeng, Jeet Sukumaran, Steven Wu and Allen Rodrigo
#
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
################################################################################


import numpy


class Individual(object):
    def __init__(self, environment, number_of_individual_species, species_registry):  # the host
        self.number_of_environmental_species = len(environment)
        self.microbiome = numpy.random.multinomial(number_of_individual_species, environment)  # initial composition is totally determined by initial environment
        self.number_of_microbes_in_host = number_of_individual_species  # how many microbes in each host
        self.species_registry = species_registry

    def __str__(self):
        self.microbiome_sequence = ['0'] * self.number_of_environmental_species
        for i in range(len(self.microbiome)):
            if self.microbiome[i] != 0:
                self.microbiome_sequence[self.species_registry.species_list[i]] = '1'
        return ''.join(self.microbiome_sequence)

